### ISSUE TITLE
Is there any way to specify the language version?

### ISSUE THREAD

**Issue by CreateAndInject** [icsharpcode/ILSpy#864]
I create a WinForm2.0 program and write this:
        Rectangle rect = new Rectangle();
        private void Form1_Load(object sender, EventArgs e)
        {
            rect.Width++;
        }

ILSpy3 give me this:
private void Form1_Load(object sender, EventArgs e)
{
	ref Rectangle S_1 = ref this.rect;
	S_1.Width += 1;
}
So I can't compile it with VS2010.

I know, ILSpy.exe can not specify the language version like Reflector, I mean : can I do it use CSharpDecompiler or any other?

**Comment by siegfriedpammer** [comment]
No, currently there is no such option. I believe, it would be difficult to implement in some cases.

**Comment by CreateAndInject** [comment]
Why? I find there are DecompilerSettings and CSharpFormattingOptions。It seems different version just set some properties as true, and some as false?

**Comment by siegfriedpammer** [comment]
Yes, but some transformations currently need to be done in every language version. In your above example, currently the decompiler needs the ref locals language feature to be able to represent the semantics of the generated code correctly. It would require more work and testing to implement every language version correctly.

**Comment by CreateAndInject** [comment]
In my example, ref local is redundant, can I forbid it through DecompilerSettings or any other?

**Comment by siegfriedpammer** [comment]
In your above example, currently the decompiler needs the ref locals language feature to be able to represent the semantics of the generated code correctly. It would require more work and testing to implement every language version correctly.

No, it's currently not possible to turn off some transforms, because they are needed to represent semantics correctly. That's why I marked this question a feature request.

**Comment by sharwell** [comment]
:memo: The highlighted situation is very similar to #839. Elimination of the local would simplify the output and also remove the use of a ref local.

**Comment by dgrunwald** [comment]
ref locals are our fallback solution when we couldn't detect the proper high-level pattern.
It doesn't make a sense to provide an option to disable them, because that would only leave you with an "unable to decompile code" error message. We'll instead need to fix all the different cases where high-level constructs are not detected correctly, so that ref locals become unnecessary.

**Comment by CreateAndInject** [comment]
The latest build(3.0.0.2869) has more problem:
private void Form1_Load(object sender, EventArgs e)
{
	ref Rectangle ? = ref this.rect;
	?.Width += 1;
}

the variable name is "?"
